### PR TITLE
#18303 refactoring to take the app key straight from the file name

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**
This comes to fulfill https://github.com/dotcms/core/issues/18303
The idea is removing the key field from the yml file and use instead of it the file name (as the app-key)

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java [hunk 1]
```diff
@@ -388,50 +387,77 @@ private void deleteSecrets(final String key, final String siteIdentifier, final
...
+     * According to https://dzone.com/articles/fileinputstream-fileoutputstream-considered-harmful
+     * that's very harmful
+     * @param file
+     * @return
+     * @throws DotDataException
+     */
+    private AppSchema readAppFile(final File file) throws DotDataException {
+        try (final InputStream inputStream = Files.newInputStream(Paths.get(file.getPath()))) {
+            return ymlMapper.readValue(inputStream, AppSchema.class);
+        }catch (Exception e){
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java:441]: Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5745882)

File: dotCMS/src/main/java/com/dotcms/security/apps/AppSchema.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        return iconUrl;
+    }
+
+    /**
+     * Tells the API if we allow any additional beside the ones already defined in the params map.
+     * @return
+     */
+    public boolean isAllowExtraParameters() {
+        return allowExtraParameters;
+    }
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/security/apps/AppSchema.java:76]: Codacy found an issue: [Avoid variables with short names like o](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5745882)

File: dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java [hunk 3]
```diff
@@ -388,50 +387,77 @@ private void deleteSecrets(final String key, final String siteIdentifier, final
...
+    /**
+     * There's a version of the method readValue on the ymlMapper which takes a file and internally creates directly a FileInputStream
+     * According to https://dzone.com/articles/fileinputstream-fileoutputstream-considered-harmful
+     * that's very harmful
+     * @param file
+     * @return
+     * @throws DotDataException
+     */
+    private AppSchema readAppFile(final File file) throws DotDataException {
+        try (final InputStream inputStream = Files.newInputStream(Paths.get(file.getPath()))) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java:439]: Codacy found an issue: [Unnecessary modifier 'final' on resource specification 'inputStream': resource specifications are implicitly final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5745882)

File: dotCMS/src/main/java/com/dotcms/security/apps/AppDescriptorImpl.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
+package com.dotcms.security.apps;
+
+import static org.apache.commons.io.FilenameUtils.*;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+import org.apache.commons.io.FilenameUtils;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/security/apps/AppDescriptorImpl.java:9]: Codacy found an issue: [Avoid unused imports such as 'org.apache.commons.io.FilenameUtils'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5745882)

File: dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java [hunk 5]
```diff
@@ -388,50 +387,77 @@ private void deleteSecrets(final String key, final String siteIdentifier, final
...
+    /**
+     * There's a version of the method writeValue on the ymlMapper which takes a file and internally creates directly a FileOutputStream
+     * According to https://dzone.com/articles/fileinputstream-fileoutputstream-considered-harmful
+     * that's very harmful
+     * @param file
+     * @return
+     * @throws DotDataException
+     */
+    private void writeAppFile(final File file, final AppSchema appSchema) throws DotDataException {
+        try (final OutputStream outputStream = Files.newOutputStream(Paths.get(file.getPath()))) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java:454]: Codacy found an issue: [Unnecessary modifier 'final' on resource specification 'outputStream': resource specifications are implicitly final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5745882)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/AppsResourceTest.java [hunk 6]
```diff
@@ -142,7 +145,8 @@ public void Test_Create_app_descriptor_Then_Create_App_Integration_Then_Delete_T
 
         final String appKey =  dataGen.getKey();
         final String fileName =  dataGen.getFileName();
-        try(final InputStream inputStream = dataGen.nextPersistedDescriptor()) {
+        final File file = dataGen.nextPersistedDescriptor();
+        try(final InputStream inputStream = Files.newInputStream(file.toPath())) {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/AppsResourceTest.java:149]: Codacy found an issue: [Unnecessary modifier 'final' on resource specification 'inputStream': resource specifications are implicitly final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5745882)

File: dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java:441]: Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5745882)

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/AppsResourceTest.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/AppsResourceTest.java:149]: Codacy found an issue: [Unnecessary modifier 'final' on resource specification 'inputStream': resource specifications are implicitly final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5745882)

File: dotCMS/src/main/java/com/dotcms/security/apps/AppSchema.java [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+    /**
+     * Any meaningful read
+     * @return
+     */
+    public String getDescription() {
+        return description;
+    }
+
+    /**
+     * an avatar URL
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/security/apps/AppSchema.java:81]: Codacy found an issue: [Avoid variables with short names like o](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5745882)

File: dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java:454]: Codacy found an issue: [Unnecessary modifier 'final' on resource specification 'outputStream': resource specifications are implicitly final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5745882)

File: dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java:439]: Codacy found an issue: [Unnecessary modifier 'final' on resource specification 'inputStream': resource specifications are implicitly final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5745882)

**Review by jdotcms — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java [hunk 8]
```diff
@@ -388,50 +387,77 @@ private void deleteSecrets(final String key, final String siteIdentifier, final
...
+     * According to https://dzone.com/articles/fileinputstream-fileoutputstream-considered-harmful
+     * that's very harmful
+     * @param file
+     * @return
+     * @throws DotDataException
+     */
+    private AppSchema readAppFile(final File file) throws DotDataException {
+        try (InputStream inputStream = Files.newInputStream(Paths.get(file.getPath()))) {
+            return ymlMapper.readValue(inputStream, AppSchema.class);
+        }catch (Exception e){
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java:441]: Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5745882)

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java [hunk 9]
```diff
@@ -389,51 +388,78 @@ private void deleteSecrets(final String key, final String siteIdentifier, final
...
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java:442]: Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5745882)

**Pull request merged**